### PR TITLE
feat(web-analytics): Add overview stats

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

Continuing web analytics

## Changes

Add a new query for some overview stats
Move shared parts of some queries into a common file
Fix an issue with the pathname cte when there weren't any pageview events
Add an extremely basic grid layout, which is still very much WIP
Screenshot 2023-10-02 at 21 14 31
(the next step will be improving the UI)

## How did you test this code?
(Behind FF) ran it

### CODE REVIEW

**Review by mariusandra — CHANGES_REQUESTED** [review]
Overall looks good, just some suggestions to get this in line with out frontend guidelines.

File: posthog/hogql_queries/web_analytics/top_sources.py [hunk 1]
```diff
@@ -16,75 +17,11 @@ class WebTopSourcesQueryRunner(WebAnalyticsQueryRunner):
     def to_query(self) -> ast.SelectQuery | ast.SelectUnionQuery:
         with self.timings.measure("top_sources_query"):
             top_sources_query = parse_select(
-                """
+                f"""
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply posthog/hogql_queries/web_analytics/top_sources.py:20]: I did this initially, 
Screenshot 2023-10-03 at 11 37 19

I get this error message `"no viable alternative at input 'WITH\\n\\nsession_cte AS (\\n{SESSION_CTE}'"`.  Am I trying to do something too crazy?

[the remark this answers is not in the document]
Comment (mariusandra) [reply posthog/hogql_queries/web_analytics/top_sources.py:20]: No need to use WITH in this case, just put the placeholder in the FROM field. That should work.

That said, this is a bug. Placeholders should/could work everywhere, but seems like they don't here.

**Review by mariusandra — APPROVED** [review]
I still believe you're over-engineering it with the CSS grid (e.g. did you think of mobile? we have had many problems with the grid on dashboards, etc...), and it would be better for everyone to just do:

```jsx
<div className='space-y-2'>
  <div className='flex flex-col gap-2'>
    <div className='w-50 sm:w-full md:w-full'>{first_one}</div>
    <div className='w-50 sm:w-full md:w-full'>{second_one}</div>
  </div>
  <div>{third one}</div>
</div>
```

or something like that and call it a day. 

Since this is feature flagged, I'll 🤷 now, but this will come back.

**Pull request merged**